cmd/history_index.cpp

```diff
@@ -0,0 +1,213 @@
+
+    std::string db_path{db::default_path()};
+    bool full, storage;
+    app.add_option("-d,--datadir", db_path, "Path to a database populated by Turbo-Geth", true)
+        ->check(CLI::ExistingDirectory);
+
+    app.add_flag("--full", full, "Start making history indexes from block 0");
+    app.add_flag("--storage", storage, "Do history of storages");
+
+
+    CLI11_PARSE(app, argc, argv);
+
+
+    // Check data.mdb exists in provided directory
+    boost::filesystem::path db_file{boost::filesystem::path(db_path) / boost::filesystem::path("data.mdb")};
+    if (!boost::filesystem::exists(db_file)) {
+        SILKWORM_LOG(LogError) << "Can't find a valid TG data file in " << db_path << std::endl;
+        return -1;
+    }
+    fs::path datadir(db_path);
+    fs::path etl_path(datadir.parent_path() / fs::path("etl-temp"));
+    fs::create_directories(etl_path);
+    etl::Collector collector(etl_path.string().c_str(), /* flush size */ 512 * kMebi);
+
+    lmdb::DatabaseConfig db_config{db_path};
+    db_config.set_readonly(false);
+    std::shared_ptr<lmdb::Environment> env{lmdb::get_env(db_config)};
+    std::unique_ptr<lmdb::Transaction> txn{env->begin_rw_transaction()};
+    // We take data from header table and transform it and put it in blockhashes table
+    lmdb::TableConfig changeset_config = storage ? db::table::kPlainStorageChangeSet : db::table::kPlainAccountChangeSet;
+    lmdb::TableConfig index_config = storage ? db::table::kStorageHistory : db::table::kAccountHistory;
+    const char * stage_key = storage ? db::stages::kStorageHistoryIndexKey : db::stages::kAccountHistoryIndexKey;
+
+    auto changeset_table{txn->open(changeset_config)};
+    std::unordered_map<std::string, roaring::Roaring64Map> bitmaps;
+
+    try {
+        auto last_processed_block_number{db::stages::get_stage_progress(*txn, stage_key)};
+
+        if (full) {
+            last_processed_block_number = 0;
+        }
+
+        // Extract
+        Bytes start(8, '\0');
+        boost::endian::store_big_u64(&start[0], last_processed_block_number + 1);
+        MDB_val mdb_key{db::to_mdb_val(start)};
+        MDB_val mdb_data;
+        if (storage) {
+            SILKWORM_LOG(LogInfo) << "Started Storage Index Extraction" << std::endl;
+        } else {
+            SILKWORM_LOG(LogInfo) << "Started Account Index Extraction" << std::endl;
+        }
+
+        size_t allocated_space{0};
+        uint64_t block_number{0};
+        int rc{changeset_table->seek(&mdb_key, &mdb_data)};  // Sets cursor to nearest key greater equal than this
+        while (!rc) {                                     /* Loop as long as we have no errors*/
+            Bytes mdb_key_as_bytes{static_cast<uint8_t*>(mdb_key.mv_data), mdb_key.mv_size};
+            auto composite_key{get_history_key(mdb_key_as_bytes)};
+            if (bitmaps.find(composite_key) == bitmaps.end()) {
+                bitmaps.insert({composite_key, roaring::Roaring64Map()});
```

Comment:
Use bitmaps.emplace instead.
Your syntax create an entry and a new one by copy contructor.